### PR TITLE
CDRIVER-3985 Remove Oppressive Terminology From Code

### PR DESCRIPTION
**Pull request by alcaeus**
CDRIVER-3985

This PR deprecates and removes oppressive terminology from code. The spec test changes will be synced one more time once mongodb/specifications#990 is merged.

~The last two commits commit are the changes from #793 and #792 to keep the build somewhat happy, so I would recommend looking at the changes commit by commit.~

### CODE REVIEW

File: src/libmongoc/doc/mongoc_server_description_ismaster.rst [hunk 1]
```diff
@@ -29,7 +29,7 @@ Parameters
 Description
 -----------
 
-The client or client pool periodically runs an `"isMaster" <https://docs.mongodb.org/manual/reference/command/isMaster/>`_ command on each server, to update its view of the MongoDB deployment. Use :symbol:`mongoc_client_get_server_descriptions()` and ``mongoc_server_description_ismaster()`` to get the most recent "isMaster" response.
+The client or client pool periodically runs a `"hello" <https://docs.mongodb.org/manual/reference/command/isMaster/>`_ command on each server, to update its view of the MongoDB deployment. Use :symbol:`mongoc_client_get_server_descriptions()` and ``mongoc_server_description_hello()`` to get the most recent "hello" response.
```
Comment (benjirewis) [inline src/libmongoc/doc/mongoc_server_description_ismaster.rst:32]: The link to the docs should be updated to https://docs.mongodb.com/manual/reference/command/hello/.

File: src/libmongoc/src/mongoc/mongoc-error-private.h [hunk 2]
```diff
@@ -45,11 +45,14 @@ typedef enum {
    MONGOC_SERVER_ERR_EXCEEDEDTIMELIMIT = 262,
    MONGOC_SERVER_ERR_SOCKETEXCEPTION = 9001,
    MONGOC_SERVER_ERR_NOTMASTER = 10107,
```
Comment (benjirewis) [inline src/libmongoc/src/mongoc/mongoc-error-private.h:47]: Do we still need these legacy codes in the enum?

File: src/libmongoc/src/mongoc/mongoc-flags.h [hunk 3]
```diff
@@ -78,7 +78,7 @@ typedef enum {
  * @MONGOC_QUERY_NONE: No query flags supplied.
  * @MONGOC_QUERY_TAILABLE_CURSOR: Cursor will not be closed when the last
  *    data is retrieved. You can resume this cursor later.
- * @MONGOC_QUERY_SLAVE_OK: Allow query of replica slave.
+ * @MONGOC_QUERY_SECONDARY_OK: Allow query of replica slave.
```
Comment (benjirewis) [inline src/libmongoc/src/mongoc/mongoc-flags.h:81]:
```suggestion
 * @MONGOC_QUERY_SECONDARY_OK: Allow query of replica secondary.
```

File: src/libmongoc/src/mongoc/mongoc-interrupt-private.h [hunk 4]
```diff
@@ -23,8 +23,8 @@
 
 /* Creates a stream to use to interrupt calls to mongoc_stream_poll.
  *
- * The expected use is to cancel in-progress ismaster commands (especially for
- * awaitable ismaster). An ismaster command may not respond for a long time, so
+ * The expected use is to cancel in-progress hello commands (especially for
+ * awaitable hello). An hello command may not respond for a long time, so
```
Comment (benjirewis) [inline src/libmongoc/src/mongoc/mongoc-interrupt-private.h:27]:
```suggestion
 * awaitable hello). A hello command may not respond for a long time, so
```

File: src/libmongoc/tests/test-mongoc-cluster.c [hunk 5]
```diff
@@ -1076,7 +1076,7 @@ _test_not_primary (bool pooled,
       BSON_ASSERT (sd->type == MONGOC_SERVER_STANDALONE);
 
       /*
-       * command error marks server Unknown iff it's a "not master" error
+       * command error marks server Unknown iff it's a "not primary" error
```
Comment (benjirewis) [inline src/libmongoc/tests/test-mongoc-cluster.c:1079]:
```suggestion
       * command error marks server Unknown if it's a "not primary" error
```

File: src/libmongoc/tests/test-mongoc-uri.c [hunk 6]
```diff
@@ -173,25 +173,6 @@ test_mongoc_uri_new (void)
    ASSERT (!bson_iter_next (&iter));
```
Comment (benjirewis) [inline src/libmongoc/tests/test-mongoc-uri.c:173]: Should the tests in this file be replaced with ones for `MONGOC_URI_SECONDARY_OK`?

**Review by benjirewis — COMMENTED** [review]
Looking good!

File: src/libmongoc/doc/mongoc_server_description_ismaster.rst (hunk 1 above)
Comment (alcaeus) [reply src/libmongoc/doc/mongoc_server_description_ismaster.rst:32]: Updated

File: src/libmongoc/src/mongoc/mongoc-error-private.h (hunk 2 above)
Comment (alcaeus) [reply src/libmongoc/src/mongoc/mongoc-error-private.h:47]: You are correct, this enum is not part of the public API. I removed the flags.

File: src/libmongoc/src/mongoc/mongoc-flags.h (hunk 3 above)
Comment (alcaeus) [reply src/libmongoc/src/mongoc/mongoc-flags.h:81]: I did a second pass over to catch those instances and fixed them.

File: src/libmongoc/src/mongoc/mongoc-interrupt-private.h (hunk 4 above)
Comment (alcaeus) [reply src/libmongoc/src/mongoc/mongoc-interrupt-private.h:27]: Did a pass through the code to catch all instances of "an hello command"

File: src/libmongoc/tests/test-mongoc-uri.c (hunk 6 above)
Comment (alcaeus) [reply src/libmongoc/tests/test-mongoc-uri.c:173]: This was removed entirely as per CDRIVER-3989; the option has been deprecated and users should use `readPreference=secondaryPreferred`, hence removing all tests for the legacy secondaryOk URI option.

**Review by benjirewis — APPROVED** [review]
LGTM!

**Review by kevinAlbs — COMMENTED** [review]
Looking great!

I ran the grep command that James included in the comment of CDRIVER-3887, I think there are a handful of docs that need updating (the other "not master" changes in code I think will be part of CDRIVER-3997):

```
$ find . -iname "*.rst" | xargs ggrep -PriI '(docs\.mongodb\.com\/master|(?<!/)master(?!.?key)|slave|whitelist|blacklist)' 
./libmongoc/tests/json/retryable_reads/README.rst:waiting time after a "not master" error:
./libmongoc/doc/installing.rst:Clone the repository and prepare the build on the current master branch or a particular release tag:
./libmongoc/doc/mongoc_server_description_t.rst:    mongoc_server_description_ismaster
./libmongoc/doc/mongoc_collection_update_many.rst:  | `MongoDB update command documentation <https://docs.mongodb.com/master/reference/command/update/>`_ for more information on the update options.
./libmongoc/doc/mongoc_collection_update_one.rst:  | `MongoDB update command documentation <https://docs.mongodb.com/master/reference/command/update/>`_ for more information on the update options.
./libmongoc/doc/mongoc_server_description_hello_response.rst:`"hello" <https://docs.mongodb.org/manual/reference/command/isMaster/>`_
./libmongoc/doc/mongoc_read_concern_t.rst:See `Read Concern Levels <https://docs.mongodb.com/master/reference/read-concern/#read-concern-levels>`_ in the MongoDB manual for more information about the individual read concern levels.
./libmongoc/doc/mongoc_client_set_appname.rst:Sets the application name for this client. This string, along with other internal driver details, is sent to the server as part of the initial connection handshake (`"hello" <https://docs.mongodb.org/manual/reference/command/isMaster/>`_).
./libmongoc/doc/mongoc_collection_replace_one.rst:  | `MongoDB update command documentation <https://docs.mongodb.com/master/reference/command/update/>`_ for more information on the update options.
```

I think docs links can be updated:
- `master` => `manual`.
- `https://docs.mongodb.com/manual/reference/command/isMaster/` => `https://docs.mongodb.com/manual/reference/command/hello/`

File: src/libmongoc/doc/mongoc_server_description_ismaster.rst [hunk 7]
```diff
@@ -1,6 +1,6 @@
-:man_page: mongoc_server_description_ismaster
+:man_page: mongoc_server_description_hello
 
-mongoc_server_description_ismaster()
+mongoc_server_description_hello()
```
[the remark this answers is not in the document]
Comment (alcaeus) [reply src/libmongoc/doc/mongoc_server_description_ismaster.rst]: Yes, this was an unintended change, good catch! I've reverted this to the legacy naming; documentation for the deprecated function should stay in place until the function itself is removed.

**Comment by alcaeus** [comment]
Yes, more changes are incoming for CDRIVER-3997, which should also take care of the last tests. I've removed the other instances shown in your comment.

**Review by kevinAlbs — APPROVED** [review]
LGTM mod one minor comment:
- README.rst has "- Examples: master branch, 1.9.5 tag". I don't think the branch helps, so I think this can be just "- Examples: 1.9.5 tag"

**Pull request merged**